gql_build/lib/src/config.dart

```diff
@@ -1,3 +1,6 @@
+const libSyntheticExtension = r"$lib$";
```

Comment:
I'd love to locate generated file next to the source file. That would require smarter importing (like, parsing all imported documents separately and passing them to builders) and type resolution logic.

Reply:
> I'd love to locate generated file next to the source file.

Yes, I agree, it'd be better if we can generate the source next to the fragment definitions.

> That would require smarter importing (like, parsing all imported documents separately and passing them to builders) and type resolution logic.

Can you expand on this?

Reply:
Currently the imported documents are concatenated to the document importing them, so that we can build a valid AST output, for example.

That approach works good enough, but adds some duplication to the generated code. An alternative approach would be to translate imports to dart imports.

`# import "some_fragment.graphql"` `->` `import "some_fragment.*.gql.dart"`

The problem is that a single `*.graphql` document can generate multiple `*.gql.dart` files. Currently that mapping is owned by `gql_build` so – how do we pass that information to `gql_code_builder`? I'm thinking it's at least a list of imports and the related `DocumentNode`s, and the mapped paths.

In the `ast_builder` we want to add all the imported definitions to the exported `document` so we must import the `*.ast.gql.dart` and traverse the AST to find the names of the definitions.

In `data_builder` we want to know which import contains the fragment to traverse it and build all the necessary getters.

To be fair, I do not fully understand the problem to give you the best explanation, but I hope this gives you some insight.

Reply:
I might be overthinking it. Maybe there's a simple "good enough" solution I do not yet see for the `frag_builder` use case.

Reply:
> To be fair, I do not fully understand the problem to give you the best explanation, but I hope this gives you some insight.

Thanks for elaborating. I think you have a better mental model than I do, but I'll think on it a bit more.